core/Inst.hpp

```diff
@@ -205,13 +219,15 @@ namespace olympia
         uint64_t               unique_id_     = 0; // Supplied by Fetch
         uint64_t               program_id_    = 0; // Supplied by a trace Reader or execution backend
         bool                   is_speculative_ = false; // Is this instruction soon to be flushed?
+        bool                   is_store_ = opcode_info_->getDestOpInfoList().size() == 0 && inst_arch_info_->isLoadStore();
```

Comment:
Actually Mavis has the information you're looking for directly:

https://github.com/sparcians/mavis/blob/c3c1060cf59baf8c16c13ddd2473f84e7c3aeb65/mavis/OpcodeInfo.h#L284C14-L287

Change this variable to a `const bool is_store_;` and do this in the _constructor_ of the instruction:
```
        Inst(const mavis::OpcodeInfo::PtrType& opcode_info,
             const InstArchInfo::PtrType     & inst_arch_info,
             const sparta::Clock             * clk) :
            opcode_info_    (opcode_info),
            inst_arch_info_ (inst_arch_info),
            is_store_(opcode_info->isInstType(mavis::OpcodeInfo::InstructionTypes::STORE),
            status_state_(Status::FETCHED)
...
```